eth/stagedsync/default_stages.go

```diff
@@ -23,7 +23,7 @@ func DefaultStages(ctx context.Context, sm prune.Mode, headers HeadersCfg, cumul
 				return nil
 			},
 			Prune: func(firstCycle bool, p *PruneState, tx kv.RwTx) error {
-				return nil
+				return retireBlocksInSingleBackgroundThread(p, senders, ctx, tx)
```

Comment:
it will work: tx==nil at initial cycle

Reply:
why stage_snapshots accepting `SendersCfg` object? it's config of stage_sender

Reply:
all stages have func which doing pruning with meaningful name like `PruneSendersStage`, why stage_snapshots also need such func

Reply:
before: this func did run "before stage_senders.prune", now it run "after stage_headers.prune" - is change order safe?